### ISSUE TITLE
Default parameters of interface type are silently set to null for string constants

### ISSUE THREAD

**Issue by grayed** [dotnet/roslyn#59789]
**Version Used**: 
4.1.0-5.22109.6 (0c82c411)

**Steps to Reproduce**:
The following minimal code demonstrates the problem:
```csharp
record Foo(IEnumerable<char> Bar = "x");
var f = new Foo();
Assert.IsNotNull(f.Bar);
```
Same happens in any method with default argument of interface type that `System.String` implements.
The usage of `System.Object` as argument type fails, though.

**Expected Behavior**:
Either method argument set to the given non-null constant value, or compiler error is raised (like for `System.Object` case right now).

**Actual Behavior**:
Method argument silently set to `null`.

**Comment by jcouv** [comment]
[Confirmed](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEAMACdBGAFgG4BYAKHPQGZcAmbAYWwG9zt3cb0DsBZACnxUAPAGMAFhABOAPmwAPbAF5sAInEBTADZaA9qoCU2Nh1ZkOFvvwOlzHAL7l7QA===):
```
using System.Collections.Generic;

public class C 
{
    public void M(IEnumerable<char> x = "hello") 
    {
        M();  // emitted as M(null)
    }
}
```

**Comment by RikkiGibson** [comment]
In case it helps: I believe it *is* expected to silently give 'null' as the argument here if the method we are calling is from metadata. However, it is important that we give an error on the parameter if the method is in source.

**Comment by RikkiGibson** [comment]
> I could not find anything in the spec (section on [optional parameters](https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/classes.md#14621-general)) saying whether this should work.

My reading of the spec is that `void M(IEnumerable<char> e = "a") { }` is disallowed.

https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/classes.md#14621-general
> The *expression* in a *default_argument* shall be one of the following:
> 
> -  a *constant_expression*
> -  an expression of the form `new S()` where `S` is a value type
> -  an expression of the form `default(S)` where `S` is a value type
> 
> The *expression* shall be implicitly convertible by an identity or nullable conversion to the type of the parameter.

https://github.com/dotnet/csharpstandard/blob/draft-v6/standard/expressions.md#1120-constant-expressions
> The following conversions are permitted in constant expressions:
> ...
> -   Implicit and explicit reference conversions, provided the source of the conversions is a constant expression that evaluates to the `null` value.

**Comment by CyrusNajmabadi** [comment]
I think your reading is correct.  Though i also find this so odd :)  I wonder why an implicit reference conversion is not ok for a non-null constant value... i'm def not sure what would be wrong with that.
